packages/client/src/pages/booking_emails/index.tsx

```diff
@@ -0,0 +1,182 @@
+    }
+  };
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = (
+    values: OrganizationData["emailTemplates"],
+    actions: FormikHelpers<OrganizationData["emailTemplates"]>
+  ) => {
+    const customers = allCustomers.filter((cus) =>
+      selectedCustomers.includes(cus.id)
+    );
+    setUpdatedEmailTemplate(values);
+    setLocalSetSubmitting(() => actions.setSubmitting);
+    openBookingsLinkDialog({
+      customers,
+      submitting: false,
+      action: "open",
+    });
+  };
+
+  const onConfirm = () => {
+    const customers = allCustomers.filter((cus) =>
+      selectedCustomers.includes(cus.id)
+    );
+
+    const monthDeadline = i18n.t(DateFormat.Deadline, {
+      date: getMonthDeadline(calendarDay),
+    });
+    localSetSubmitting(true);
+
+    dispatch(
+      updateOrganizationEmailTemplates(updatedEmailTemplate, localSetSubmitting)
+    );
+
+    customers.forEach((customer) => {
+      dispatch(
+        sendBookingsLink({
+          ...customer,
+          method: ClientMessageMethod.Email,
+          bookingsLink: getBookingsLink(customer.secretKey),
+          deadline: monthDeadline,
+        })
+      );
+    });
+    close();
+    localSetSubmitting(false);
+  };
+
+  const onCancel = () => {
+    localSetSubmitting(false);
+    close();
+  };
+
+  useEffect(() => {
+    state &&
+      state.props &&
+      state.props.submitting &&
+      (state.props.action === "confirm" ? onConfirm() : onCancel());
+  }, [state]);
+
+  const initialValues = {
+    ...emailTemplates,
+    ...organization.emailTemplates,
+  } as OrganizationData["emailTemplates"];
+
+  return (
+    <Layout>
+      <Formik
+        {...{ initialValues }}
+        enableReinitialize={true}
+        onSubmit={(values, actions) => handleSubmit(values, actions)}
+      >
+        {({ isSubmitting, isValidating, handleReset }) => (
+          <>
+            <LayoutContent
+              Component={Form as any}
+              actionButtons={
+                <div className="py-2 flex justify-end items-center gap-2">
+                  <Button
+                    onClick={handleReset}
+                    disabled={isSubmitting || isValidating}
+                    className="!text-cyan-500"
+                    size={ButtonSize.MD}
+                  >
+                    {t(ActionButton.Reset)}
+                  </Button>
+                  <Button
+                    disabled={
+                      !selectedCustomers.length || isSubmitting || isValidating
+                    }
+                    color={ButtonColor.Primary}
+                    size={ButtonSize.MD}
+                    aria-label={"save"}
+                    type="submit"
+                  >
+                    {t(ActionButton.Save)}
+                  </Button>
+                </div>
+              }
+            >
+              <EmailTemplateSettings
+                onCheckboxChange={handleCheckboxChange}
+                customers={allCustomers}
```

Comment:
This should be `nonDeletedCustomers`, not `allCustomers`. I.e. we need to filter out deleted customers.